pulsar-client-go/pulsar/schema.go

```diff
@@ -0,0 +1,306 @@
+	schemaInfo := NewSchemaInfo("String", "", STRING)
+	strSchema := &StringSchema{
+		*schemaInfo,
+	}
+	return strSchema
+}
+
+func (ss *StringSchema) Serialize(data string) []byte {
+	sh := (*reflect.StringHeader)(unsafe.Pointer(&data))
+	bh := reflect.SliceHeader{
+		Data: sh.Data,
+		Len:  sh.Len,
+		Cap:  sh.Len,
+	}
+	return *(*[]byte)(unsafe.Pointer(&bh))
+}
+
+func (ss *StringSchema) UnSerialize(data []byte) string {
+	bh := (*reflect.SliceHeader)(unsafe.Pointer(&data))
+	sh := reflect.StringHeader{
+		Data: bh.Data,
+		Len:  bh.Len,
+	}
+	return *(*string)(unsafe.Pointer(&sh))
+}
+
+type BytesSchema struct {
+	SchemaInfo
+}
+
+func NewBytesSchema() *BytesSchema {
+	schemaInfo := NewSchemaInfo("Bytes", "", BYTES)
+	bytesSchema := &BytesSchema{
+		*schemaInfo,
+	}
+	return bytesSchema
+}
+
+func (bs *BytesSchema) Serialize(data []byte) []byte {
+	return data
+}
+
+func (bs *BytesSchema) UnSerialize(data []byte) []byte {
+	return data
+}
+
+type Int8Schema struct {
+	SchemaInfo
+}
+
+func NewInt8Schema() *Int8Schema {
+	schemaInfo := NewSchemaInfo("INT8", "", INT8)
+	int8Schema := &Int8Schema{
+		*schemaInfo,
+	}
+	return int8Schema
+}
+
+func (bs *Int8Schema) Serialize(value int8) ([]byte, error) {
+	var buf bytes.Buffer
+	err := WriteElements(&buf, value)
+	return buf.Bytes(), err
+}
+
+func (bs *Int8Schema) UnSerialize(data []byte, value int8) error {
+	buf := bytes.NewReader(data)
+	return ReadElements(buf, value)
+}
+
+type Int16Schema struct {
+	SchemaInfo
+}
+
+func NewInt16Schema() *Int16Schema {
+	schemaInfo := NewSchemaInfo("INT16", "", INT16)
+	int16Schema := &Int16Schema{
+		*schemaInfo,
+	}
+	return int16Schema
+}
+
+func (bs *Int16Schema) Serialize(value int16) ([]byte, error) {
+	var buf bytes.Buffer
+	err := WriteElements(&buf, value)
+	return buf.Bytes(), err
+}
+
+func (bs *Int16Schema) UnSerialize(data []byte, value int16) error {
+	buf := bytes.NewReader(data)
+	return ReadElements(buf, value)
+}
+
+type Int32Schema struct {
+	SchemaInfo
+}
+
+func NewInt32Schema() *Int32Schema {
+	schemaInfo := NewSchemaInfo("INT32", "", INT32)
+	int32Schema := &Int32Schema{
+		*schemaInfo,
+	}
+	return int32Schema
+}
+
+func (bs *Int32Schema) Serialize(value int32) ([]byte, error) {
+	var buf bytes.Buffer
+	err := WriteElements(&buf, value)
+	return buf.Bytes(), err
+}
+
+func (bs *Int32Schema) UnSerialize(data []byte, value int32) error {
+	buf := bytes.NewReader(data)
+	return ReadElements(buf, value)
+}
+
+type Int64Schema struct {
+	SchemaInfo
+}
+
+func NewInt64Schema() *Int64Schema {
+	schemaInfo := NewSchemaInfo("INT64", "", INT64)
+	int64Schema := &Int64Schema{
+		*schemaInfo,
+	}
+	return int64Schema
+}
+
+func (bs *Int64Schema) Serialize(value int64) ([]byte, error) {
+	var buf bytes.Buffer
+	err := WriteElements(&buf, value)
+	return buf.Bytes(), err
+}
+
+func (bs *Int64Schema) UnSerialize(data []byte, value int64) error {
+	buf := bytes.NewReader(data)
+	return ReadElements(buf, value)
+}
+
+type Float32Schema struct {
```

Comment:
Let's use `Float` for `Float32` and `Double` for `Float64` in the remaining places.

Reply:
yes , will fix it